assets/src/js/Components/ArchivePicker.js

```diff
@@ -1,125 +1,274 @@
-    <form
-      className="archive-picker-search"
-      onSubmit={event => {
-        event.preventDefault();
-        if (!loading) {
-          setSearchText(enteredSearch);
-          setCurrentPage(1);
-        }
+  useEffect(() => {
+    if (pageNumber > 1 && !loading && !loaded) {
+      setAbortController(new AbortController());
+    }
+  }, [pageNumber]);
+
+  useEffect(() => {
+    if (showAddedMessage) {
+      setTimeout(() => {
+        dispatch({type: 'HIDE_ADDED_MESSAGE'});
+      }, 5000);
+    }
+  }, [showAddedMessage]);
+
+  return useMemo(() => (
+    <Context.Provider
+      value={{
+        loading,
+        loaded,
+        error,
+        fetch,
+        images,
+        processingError,
+        processingImages,
+        showAddedMessage,
+        includeInWp,
+        selectedImages,
+        searchText,
+        dispatch,
       }}
-      onChange={event => setEnteredSearch(event.target.value)}
     >
-      <input
-        type="text"
-        disabled={loading}
-      />
-      <button
-        disabled={loading}
-        type="submit"
-        className="button"
-      >{__('Search', 'planet4-master-theme-backend')}</button>
-    </form>
-    {loading && (
-      <div className="archive-picker-loading">{__('Loading...', 'planet4-master-theme-backend')}</div>
-    )}
-    {!!error && (
-      <div>
-        <h3>API error:</h3>
-        <p> {error.message} </p>
-      </div>
-    )}
-    <div className={classNames('image-picker', {'open-sidebar': selectedImages.length > 0})}>
-      <ul
-        className="picker-list"
-        onScroll={event => {
-          if (!loading && isNearScrollEnd(event)) {
-            setCurrentPage(currentPage + 1);
-          }
-        }}
-      >
-        <ArchivePickerList
-          isSelected={isSelected}
-          toggleMultiSelection={toggleMultiSelection}
-          toggleSingleSelection={toggleSingleSelection}
-          images={images}
-        />
-      </ul>
-      {selectedImages.length > 0 && (
-        <div className={classNames('picker-sidebar', {'picker-sidebar-single': selectedImages.length === 1})}>
-          {selectedImages.length === 1 ?
-            <SingleSidebar
-              image={selectedImages[0]}
-              processingError={processingError}
-              processingImages={processingImages}
-              includeInWp={includeInWp}
-              closeSidebar={clearSelectedImages}
-              showAddedMessage={showAddedMessage}
-            /> :
-            <MultiSidebar
-              selectedImages={selectedImages}
-              toggleMultiSelection={toggleMultiSelection}
-            />
-          }
+      <MultiSearchOption />
+
+      {!!error && (
+        <div>
+          <h3>API error:</h3>
+          <p> {error.message} </p>
         </div>
       )}
-    </div>
-  </Fragment>;
-};
 
-export default ArchivePicker;
+      <div className={classNames('image-picker', {'open-sidebar': Object.keys(selectedImages).length})}>
```

Comment:
Maybe you could create a `numberOfSelectedImages` variable or something like that instead of duplicating several times the `Object.keys(selectedImages).length` line? I think it would be a bit easier to read

Reply:
Totally agree!